docs/en/observability/ci-cd-observability.asciidoc

```diff
@@ -427,3 +427,58 @@ The Concourse CI doesn’t report health metrics through OpenTelemetry. However,
 to derive pipeline execution traces into KPI metrics like throughput and the error rate
 of pipelines.
+
+
+[discrete]
+[[check-service-health-from-cd-pipelines]]
+== Check service health from deployment pipelines
+
+
+Integrating automated service health checks in deployment pipelines is critical for end-to-end deployment automation, which crucially enables deployment frequency to be increased.
+
+Elastic Observability exposes service HTTP health check APIs that can be integrated in deployment pipelines to verify the behaviour of newly deployed instances, and either automatically continue the deployments or rollback according to the health status.
+
+The following example shows a canary deployment pipeline that leverages Elastic health check HTTP APIs to automate the quality check before rolling out the deployment from the canary to the entire set of instances:
+
+image::images/ci-cd-canary-deployment-pipeline.png[Canary Deployment Pipeline]
+
+
+The primary HTTP API to check the health of a service is compare the transaction error rate with a threshold value.
+
+The API call requires authentication, we recomend to use an API Token to authenticate.
+
+The URL is `KIBANA_URL/internal/apm/services`, it is designed to query service metrics.
+
+In order to check the transaction error rate of the service on the newly deployed instance, we have to pass the following parameters to the API call:
+
+* `start` and `end`: time interval using the https://en.wikipedia.org/wiki/ISO_8601[ISO-8601] format (e.g. "2021-09-01T13:24:12Z" which is an UTC time)
+* `kuery`: used to filter on the service name and for example the new version being deployed to narrow down to the canary instances. Example `service.name:"MY_SERVICE_NAME" AND service.version:"1.2.3"`
+* `environment`: the environment on which the canary instances are deployed. Example: `production`.
+
+We need a range of time, this range is defined by the parameters `start` and `end`, 
+these parameters are two dates in ISO-8601 format. Decause we want to query only one service,
+we have to compose a filter in the param `kuery`, we can filter by the service name using the expresion `service.name:MY_SERVICE_NAME and srevive.version: SERVICE_VERSION`.
+Finally we apply a filter by environment passing the parameter `environment`, 
+to select all environments the value is `ENVIRONMENT_ALL`
+
+The API is subject to changes and a stable API optimized for Continous Delivery use cases will soon be published.
+
+[source,python]
+----
+def check_service_health(service_name, service_version, error_rate_threshold, kibana_url, api_token):
+    now = datetime.now()
+    five_minutes_ago = now - timedelta(minutes=5)
+    params = {
+        "start": five_minutes_ago.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "end": now.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "kuery": "service.name:{} and service.version:{}".format(service_name, servive_version),
```

Comment:
```suggestion
        "kuery": "service.name:{} and service.version:{}".format(service_name, service_version),
```

Reply:
I assume this is a typo. Can you confirm that this example has been tested please? 

Also is kuery really spelled with a "k"?